### PR TITLE
Add fourth section of DDD docs

### LINKED ISSUE #1266: Write the fourth sub-section of the DDD user guide: What happens when I use DDD?
Issue Description
-----------------
Write the fourth sub-section of the DDD user guide, namely:

4. What happens when I use DDD?

This is a sub-issue of #1210. 

Proposed Solution and additional information
-----------------
See #1210.

### PR DESCRIPTION
**Pull request by andreamari**
Description
-----------
Adds fourth section of DDD docs. Fixes #1266.

License
-------

- [x] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

### CODE REVIEW

**Comment by andreamari** [comment]
Docs preview: https://mitiq--1276.org.readthedocs.build/en/1276/guide/ddd-4-low-level.html

Tests are failing due to broken links unrelated to this PR. They are addressed in #1282.

File: docs/source/guide/ddd-4-low-level.myst [hunk 1]
```diff
@@ -12,3 +12,163 @@ kernelspec:
...
+
+```{figure} ../img/ddd_workflow.png
+---
+width: 700px
+name: ddd-workflow
+---
+Workflow of the DDD technique in Mitiq.
+```
+
+- The user provides a `QPROGRAM`, (i.e. a quantum circuit defined via any of the supported frontends).
```
Comment (Misty-W) [inline docs/source/guide/ddd-4-low-level.myst:26]: Might be good to add a link to `/frontends-backends.myst`.

File: docs/source/guide/ddd-4-low-level.myst [hunk 2]
```diff
@@ -12,3 +12,163 @@ kernelspec:
...
+This matrix contains the same amount of information as `mask_matrix`, but it is more convenient for the analysis
+of idle windows as shown in the next code cell.
+
+```{code-cell} ipython3
+import numpy as np 
+
+print(f"The circuit contains {np.count_nonzero(slack_matrix)} slack windows.")
+print(f"The maximum slack length is {np.max(slack_matrix)}.")
+lengths, counts = np.unique(slack_matrix, return_counts=True)
+lenght_distribution = dict(zip(lengths[1:], counts[1:]))
```
Comment (Misty-W) [inline docs/source/guide/ddd-4-low-level.myst:107]:
```suggestion
length_distribution = dict(zip(lengths[1:], counts[1:]))
```

File: docs/source/guide/ddd-4-low-level.myst [hunk 3]
```diff
@@ -12,3 +12,163 @@ kernelspec:
...
+print(f"The full distribution of slack lengths is {lenght_distribution}.")
```
Comment (Misty-W) [inline docs/source/guide/ddd-4-low-level.myst:108]:
```suggestion
print(f"The full distribution of slack lengths is {length_distribution}.")
```

**Review by Misty-W — COMMENTED** [review]
@andreamari the PR looks good, just a few minor suggestions.

File: docs/source/guide/ddd-4-low-level.myst (hunk 1 above)
Comment (andreamari) [reply docs/source/guide/ddd-4-low-level.myst]: Good idea! Done: https://github.com/unitaryfund/mitiq/pull/1276/commits/d588fbcc416ef358520e9fe47ea5e6e083225042

**Review by Misty-W — APPROVED** [review]

**Comment by andreamari** [comment]
Reopened after erroneously deleting the branch :-)

**Pull request merged**